src/main/java/spireCafe/abstracts/AbstractArticle.java

```diff
@@ -0,0 +1,117 @@
+package spireCafe.abstracts;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.helpers.FontHelper;
+import com.megacrit.cardcrawl.helpers.Hitbox;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.helpers.TipHelper;
+import com.megacrit.cardcrawl.helpers.input.InputHelper;
+
+public abstract class AbstractArticle {
+
+    private static final float HOVERED_SCALE = 1.1f;
+    private static final float PRICE_OFFSET = 75f;
+
+    public String articleId;
+    public AbstractMerchant merchant;
+
+    public Hitbox hb;
+    public boolean isGettingClicked = false;
+    public float xPos;
+    public float yPos;
+
+    private float scale;
+
+    public TextureRegion itemTexture;
+
+    public int price = -1;
+    public Texture priceIcon;
+
+    public AbstractArticle(String id, AbstractMerchant merchant, float x, float y, float hbWidth, float hbHeight) {
+        articleId = id;
+        this.merchant = merchant;
+        xPos = x;
+        yPos = y;
+        hb = new Hitbox(x, y, hbWidth, hbHeight);
+    }
+
+    public AbstractArticle(String id, AbstractMerchant merchant, float x, float y, Texture itemTexture) {
+        articleId = id;
+        this.merchant = merchant;
+        xPos = x;
+        yPos = y;
+        this.itemTexture = new TextureRegion(itemTexture);
+        hb = new Hitbox(this.itemTexture.getRegionWidth(), this.itemTexture.getRegionHeight());
+    }
+
+
+    public abstract boolean canBuy();
+
+    public abstract void onBuy();
```

Comment:
Would appreciate some comments for methods like these and rollShop. We talked about this so we understand, but somebody else might struggle a bit